### PR TITLE
Update date component html

### PR DESCRIPTION
**Pull request by lennym**
To bring in line with https://design-system.service.gov.uk/components/date-input/

### CODE REVIEW

File: src/forms/date.jsx [hunk 1]
```diff
@@ -21,25 +21,37 @@ class DateInput extends Input {
...
-        <legend>
-          <span className="form-label-bold">{this.props.label}</span>
-          { this.props.hint && <span className="form-hint">{this.props.hint}</span> }
-          { this.props.error && <span className="error-message">{this.props.error}</span> }
+    return <div className={this.errorClass('govuk-form-group')}>
+      <fieldset className="govuk-fieldset" aria-describedby={this.dateFragment('hint')} role="group">
+        <legend className="govuk-fieldset__legend">
+          <span className="govuk-label">{this.props.label}</span>
+          { this.props.hint && <span className="govuk-hint" id={this.dateFragment('hint')}>{this.props.hint}</span> }
+          { this.props.error && <span className="govuk-error-message">{this.props.error}</span> }
```
Comment (wheelsandcogs) [inline src/forms/date.jsx]: I think the hint and error messages probably want to be outside of the legend, but there isn't really any definitive answer here: https://design-system.service.gov.uk/components/fieldset/

File: src/forms/date.jsx [hunk 2]
```diff
@@ -21,25 +21,37 @@ class DateInput extends Input {
...
-    return <div className={this.errorClass('form-group')}>
-      <fieldset>
-        <legend>
-          <span className="form-label-bold">{this.props.label}</span>
-          { this.props.hint && <span className="form-hint">{this.props.hint}</span> }
-          { this.props.error && <span className="error-message">{this.props.error}</span> }
+    return <div className={this.errorClass('govuk-form-group')}>
+      <fieldset className="govuk-fieldset" aria-describedby={this.dateFragment('hint')} role="group">
+        <legend className="govuk-fieldset__legend">
+          <span className="govuk-label">{this.props.label}</span>
```
Comment (wheelsandcogs) [inline src/forms/date.jsx]: Do we need the span if we have the legend? For `RadioGroup` I used:
```
<legend className="govuk-fieldset__legend">
    <h2 className="govuk-fieldset__heading govuk-heading-l">{this.props.label}</h2>
</legend>
```

File: src/forms/date.jsx (hunk 1 above)
Comment (lennym) [reply src/forms/date.jsx]: I literally just copied the html from https://design-system.service.gov.uk/components/date-input/

Comment (wheelsandcogs) [reply src/forms/date.jsx]: Ah yeah they're definitely outside legend then:
```
<legend class="govuk-fieldset__legend govuk-fieldset__legend--xl">
      <h1 class="govuk-fieldset__heading">
        What is your date of birth?
      </h1>
    </legend>
    <span id="dob-hint" class="govuk-hint">
      For example, 31 3 1980
    </span>
    <span id="dob-error" class="govuk-error-message">
      Date of birth must be in the past
    </span>
```

**Pull request merged**